### ISSUE TITLE
New patch causes client to freeze up entire system

### ISSUE THREAD

**Issue by zeparu** [ValveSoftware/Dota-2#2689]
#### Your system information

* System information from steam (`Steam` -> `Help` -> `System Information`) in a [gist](https://gist.github.com/):
https://gist.github.com/zeparu/7f4d941e5455f8efd3a7ff9bc1dd3c38
 If on Linux, please include Steam -> Help -> Steam Runtime Diagnostics in the gist.
https://gist.github.com/zeparu/df84779ec0d512735959b4bd2e944cee
* Have you checked for system updates?: [Yes/No]
yes, am on current version Arch Linux
* Are you using the latest stable video driver available for your system? [Yes/No]
Yes
* Have you verified the game files?: [Yes/No]
No

#### Please describe your issue in as much detail as possible:
Describe what you _expected_ should happen and what _did_ happen. Please link any large pastes as a [Github Gist](https://gist.github.com/).

Entire System freezes up when browsing the new event interface in the client, the type of system freeze where the last second of audio loops indefinitely 

i tried searching through my `journald` log, to see if any errors stand out, but i cant see anything in there

#### Steps for reproducing this issue:

1. Open Dota 2
2. Browse the Event Pages, like the new not-cavern-crawl or the shop options
3. system will freeze after a couple minutes max

i created a subreddit post at first, will cross link it here
https://www.reddit.com/r/DotA2/comments/1c7vemn/entire_system_freezes_up_since_the_patch_on_arch/
and this issue on the post as well, i havent seen anybody else describe issues for linux, just posts about frequent game crashes on the windows version

edit: i verified the files for dota 2, and playing around cautiously in the menu, and wasnt able to re-create a crash right now, having an eye on htop to monitor memory and cpu usage; it felt like the 3d models for the 2 new arcanar/other models of other characters were lagging quite a bit before my system froze up, but right now everything seems smooth

**Comment by AngryPenguinPL** [comment]
I have same issue. Looks like dota and steam take whole of my RAM (20GB) and this cause all problems. SSD usage was also very high (~120GB data writed??) Maybe that shader compilation? Not sure. After game closing and next PC reboot game start works fine (at least main menu, not playing any matches so far.).

**Comment by zeparu** [comment]
i've been playing a couple games since i had this issue, and i haven't had a crash/freeze like that since then, maybe it was shader compilation for all the fancy menus, and now that its done, its running fine again
i suspected something like that too, which is why i checked cpu/ram usage on my attempt to make it freeze for the third time, but since then it hasnt happened again

**Comment by Ori-Pixel** [comment]
Having the same issue. Got an abandon after crashing about 30 times in a single match, pretty fucking awful experience. I normally got only a single crash a game until the patch around 3pm Seattle time. idk what was in that 130MB but it FUCKED my system badly.

Also, thanks to the changes from a few years ago, there are no dump files to upload. 

All I know is you can reproduce it very easily and instantly by demo mode terrorblade and have an enemy, spawn 3 illusions, click select all. Crashes every time.

Here is a speedrun video of forcing the crash:
[output.webm](https://github.com/ValveSoftware/Dota-2/assets/54956527/723c8f18-d498-40b5-8361-72862182c959)

Edit: for some reason this only works with terror lade for me? At least Naga illusions didn't trigger it.

**Comment by jaredleishman** [comment]
@Ori-Pixel Game crashes for me anytime I cast Meta with TB, since this latest patch.

**Comment by zeparu** [comment]
just had another hard freeze, also the game is now not even able to hit target fps of 144, even after restarting

**Comment by zeparu** [comment]
also, it doesnt have anything to do with RAM getting full, i monitored my ram and it didnt go even over 20GB (of 32GB) for it to freeze my system

**Comment by zeparu** [comment]
i can also reproduce the terrorblade crash, but its just a game crash, not a system freeze

**Comment by Ori-Pixel** [comment]
> i can also reproduce the terrorblade crash, but its just a game crash, not a system freeze

Yeah I found out the TB issue is actually a game crash, but I lose control of the whole monitor until Dota full crashes out (sometimes 30s)

**Comment by AngryPenguinPL** [comment]
Because we talking about two separate problems. One with I think shader processing at first game launch and seconds with in game terrorblade crash.

**Comment by zeparu** [comment]
have actually gotten this same freeze now without dota/steam running, just playing a movie on VLC, idk if its maybe a kernel update or something, ill try out some more stuff with linux-lts and check back

**Comment by JoonasPitk** [comment]
> just playing a movie on VLC, idk if its maybe a kernel update or something, ill try out some more stuff with linux-lts and check back

I had the same problem, with mpv. Total system freeze, not even REISUB responds. Managed to eliminate this by disabling Adaptive Sync under Display Settings in Plasma.

**Comment by zeparu** [comment]
this might be a kde issue in the end: https://bugs.kde.org/show_bug.cgi?id=485730
matches my issues so far

**Comment by regulator-g** [comment]
I'm still getting a whole system freeze, 

-Arch up to date
-plasma 6.0.5
-AMD 6700xt

originally the game wouldn't launch then tried clearing shader cache, then had to use -vulkan_disable_pipeline_cache and it managed to start ok, but its now freezing during picking phase or first 5mins of game, locks the whole system up

doesn't seem to happen with flatpak steam/dota

weirdly its not crashing when i play with bots to replicate the issue!

will try turn adaptive sync off 

if it keeps happening ill capture some logs
